src/cephadm/cephadm.py

```diff
@@ -4818,15 +4828,22 @@ def command_agent(ctx: CephadmContext) -> None:
 
 ##################################
 
-
-@infer_image
+@executes_early
 def command_version(ctx):
     # type: (CephadmContext) -> int
-    c = CephContainer(ctx, ctx.image, 'ceph', ['--version'])
-    out, err, ret = call(ctx, c.run_cmd(), desc=c.entrypoint)
-    if not ret:
-        print(out.strip())
-    return ret
+    import importlib
+
+    try:
+        vmod = importlib.import_module('_version')
+    except ImportError:
+        print('cephadm version UNKNOWN')
+        return 1
+    print('cephadm version {0} ({1}) {2} ({3})'.format(
+        vmod.CEPH_GIT_NICE_VER,
+        vmod.CEPH_GIT_VER,
+        vmod.CEPH_RELEASE_NAME,
+        vmod.CEPH_RELEASE_TYPE))
```

Comment:
assuming these correlate to the 4 things printed after `ceph version` in
```
ceph version 17.2.6-129-g2e256435 (2e256435f93b3e579546bf2426324b546c557abc) quincy (stable)
```
right? I think I remember you saying these args were getting setup similar to how the version is added for the main ceph stuff.

Reply:
Yes, this line was actually copy-pasted from `ceph.in` and then I added the `vmod.` prefix to the lines because we don't overwrite the contents of cephadm like when `ceph.in` becomes `ceph` we add a new file to the zipapp and import it.

Reply:
We *can* output whatever the heck we want here... we could print the version info in JSON for example :-)
I just copied what `ceph` does because I figured it might be familiar to people.

Reply:
I think `--format=json` would be nice but I don't think it's needed for this PR.  I don't think PEP applies because ceph has it's own unique version numbering (and naming) approach.